File: src/commands/info.py
import os
import discord
from discord.ext import commands
from discord import app_commands, Interaction, Embed
import sys
from pathlib import Path

# Add the parent directory to sys.path to allow imports
sys.path.insert(0, str(Path(__file__).parent.parent.parent))

from src.config import load_config

# Load configuration
config = load_config(os.getenv("CONFIG_PATH", "config.yaml"))
BOT_DESCRIPTION = config.get("bot", {}).get("description", "A powerful and fun Discord radio bot inspired by Alastor.")

class InfoCog(commands.Cog):
    def __init__(self, bot: commands.Bot):
        self.bot = bot

    @app_commands.command(name="info", description="Show information about the bot.")
    async def show_info(self, interaction: Interaction):
        embed = Embed(
            title="Alastor - The Radio Daemon",
            description=BOT_DESCRIPTION,
            color=discord.Color.purple()
        )
        version = config.get("bot", {}).get("version", "Unknown")
        github_url = config.get("bot", {}).get("github_url", "https://github.com/bnfone/discord-bot-alastor")
        embed.add_field(name="Version", value=version, inline=False)
        embed.add_field(name="Developer", value="[Blake](https://github.com/bnfone)", inline=False)
        embed.add_field(name="Source Code", value=f"[GitHub]({github_url})", inline=False)
        embed.set_footer(text="Alastor - The Radio Daemon")
        await interaction.response.send_message(embed=embed)

async def setup(bot: commands.Bot):
    await bot.add_cog(InfoCog(bot))

File: src/commands/info.rs
use serenity::{
    all::{
        CommandInteraction, CreateCommand, CreateInteractionResponse, 
        CreateInteractionResponseMessage, CreateEmbed, Colour,
    },
    prelude::Context,
};

use crate::{error::Result, radio::RadioManager};

pub fn create_info_command() -> CreateCommand {
    CreateCommand::new("info")
        .description("Show information about Alastor bot")
}

pub async fn handle_info_command(ctx: &Context, command: &CommandInteraction) -> Result<()> {
    let data = ctx.data.read().await;
    let radio_manager = data.get::<RadioManager>();
    
    let active_streams = radio_manager
        .map(|rm| rm.get_active_streams_count())
        .unwrap_or(0);

    let embed = CreateEmbed::new()
        .title("🎭 Alastor - The Radio Daemon")
        .description("High-performance Discord radio bot inspired by Hazbin Hotel")
        .color(Colour::PURPLE)
        .field("Version", "2.0.0 (Rust)", true)
        .field("Active Streams", active_streams.to_string(), true)
        .field("Language", "Rust 🦀", true)
        .field("Developer", "[Blake](https://github.com/bnfone)", false)
        .field("Source Code", "[GitHub](https://github.com/bnfone/discord-bot-alastor)", false)
        .field(
            "Features",
            "• High-performance concurrent audio streaming\n• Fuzzy search with autocomplete\n• Stream health checking\n• Ephemeral responses\n• Setup wizard",
            false
        )
        .thumbnail("https://i.imgur.com/your-alastor-image.png") // You can add the alastor.jpg here
        .footer(|f| f.text("Built with Serenity & Songbird"));

    command.create_response(
        &ctx.http,
        CreateInteractionResponse::Message(
            CreateInteractionResponseMessage::new()
                .embed(embed)
                .ephemeral(true)
        )
    ).await?;

    Ok(())
}